instruqt-tracks/vault-auth-aws/track.yml

```diff
@@ -0,0 +1,309 @@
+    Be sure to name the role `my-role-iam`, and for the sake of simplicity, you
+    should configure the `bound_iam_pricipal_arn` to be a wildcard for all principals
+    in your provided AWS account.
+
+    ```
+    vault write auth/aws/role/my-role-iam \
+      auth_type=iam \
+      bound_iam_principal_arn=arn:aws:iam::$AWS_ACCOUNT_ID:* \
+      policies=developer \
+      token_ttl=30m \
+      token_max_ttl=30m
+    ```
+
+    You should also configure the `iam_server_id_header_value`, to help prevent
+    replay attacks.
+
+    ```
+    vault write auth/aws/config/client \
+      iam_server_id_header_value=instruqt.vault.example.com
+    ```
+
+    Vault is now successfully configured for AWS IAM authentication.
+  notes:
+  - type: text
+    contents: |-
+      When enabled, auth methods are similar to secrets engines. They are
+      mounted within the Vault mount table and can be accessed and configured
+      using the standard read/write API. All auth methods are mounted underneath
+      the `auth/` prefix.
+  - type: text
+    contents: |-
+      The [AWS STS API](https://docs.aws.amazon.com/STS/latest/APIReference/Welcome.html)
+      includes a method, [`sts:GetCallerIdentity`](https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html),
+      which allows you to validate the identity of a client. The client signs a
+      `GetCallerIdentity` query using the AWS Signature v4 algorithm and submits
+      4 pieces of information to the Vault server to recreate a valid signed
+      request (the request URL, the request body, the request headers, and the
+      request method). The AWS signature is computed over those fields. The Vault
+      server then reconstructs the query and forwards it on to the AWS STS service
+      and validates the result back.
+  - type: text
+    contents: |-
+      Each signed AWS request includes the current timestamp to mitigate the risk
+      of replay attacks. In addition, Vault allows you to require an additional
+      header, `X-Vault-AWS-IAM-Server-ID`, to be present to mitigate against
+      different types of replay attacks (such as a signed `GetCallerIdentity`
+      request stolen from a dev Vault instance and used to authenticate to a prod
+      Vault instance).
+
+      Vault further requires that this header be one of the headers included in
+      the AWS signature and relies upon AWS to authenticate that signature.
+  tabs:
+  - title: Configure the AWS Auth Method
+    type: terminal
+    hostname: vault-server
+  - title: Vault UI
+    type: service
+    hostname: vault-server
+    port: 8200
+  difficulty: basic
+  timelimit: 300
+- slug: login-aws-auth-method
+  id: cgaexaj1xjpe
+  type: challenge
+  title: Log In Using the AWS IAM Auth Method
+  teaser: Generate a signed AWS API request that will validate against AWS and log
+    you into Vault.
+  assignment: |-
+    You are ready to perform the login operation.
+
```

Comment:
I recommend adding comment suggesting that user not export their root token at the beginning of this challenge since doing so leads to warning when logging in with AWS auth method.